### DISCUSSION TITLE
SQLBoiler + Snowflake

### DISCUSSION THREAD

**Discussion by rtjoseph11** [aarondl/sqlboiler#956] [General]
I'm potentially interested in implementing a Snowflake driver for SQLBoiler for my job and was wondering if anyone has tried this or knows that it won't be possible. The snowflake driver works with the `database/sql` so I assume I should be able to add a driver for it in SQLBoiler. Also if you have anywhere you could point me to for what needs to be done for adding new drivers that would be much appreciated.

**Comment by aarondl** [comment]
Hi @rtjoseph11. I don't really know anything about snowflake but the process of adding a driver is -mostly- not terrible. You start a new repository called sqlboiler-snowflake (the name prefix matters) and basically copy the layout of one of the existing drivers:
https://github.com/volatiletech/sqlboiler/tree/master/drivers/sqlboiler-psql

The drivers communicate over JSON with the main sqlboiler binary. You only need to implement one thing: https://github.com/volatiletech/sqlboiler/blob/master/drivers/interface.go#L29

See how psql driver does this as an example. That's mostly it. A more contained example would be https://github.com/volatiletech/sqlboiler-sqlite3

These can all be separate repos so you can experiment on your own which is why it's architected like this.
